cadence/freshmint-metadata-views/FreshmintMetadataViews.cdc

```diff
@@ -13,4 +15,49 @@ pub contract FreshmintMetadataViews {
             self.hash = hash
         }
     }
+
+    /// Create an instance of MetadataViews.IPFSFile from the provided string.
+    ///
+    /// If the string is in IPFS directory format, this function splits it
+    /// into the its CID and path.
+    ///
+    pub fun ipfsFile(file: String): MetadataViews.IPFSFile {
+        // Find the first "/" character if one exists
+        let maybeIndex = self.firstIndexInString(file, of: "/")
+
+        if let index = maybeIndex {
+            // The CID is all characters before the "/"
+            let cid = file.slice(from: 0, upTo: index)
+
+            // The remaining characters are the IPFS path
+            let path = file.slice(from: index + 1, upTo: file.length)
+
+            return MetadataViews.IPFSFile(
+                cid: cid,
+                path: path
+            )
+        }
+
+        return MetadataViews.IPFSFile(
+            cid: file,
+            path: nil
+        )
+    }
+
+    /// Return the first index of a character in the string,
+    /// or nil if none exists.
+    ///
+    pub fun firstIndexInString(_ s: String, of: Character): Int? {
+        var i = 0
+
+        while i < s.length {
```

Comment:
will it throw if "s" is nil or in cadence it will not?

Reply:
Cadence will guarantee that `String` is never nil because it's a non-optional type. If the type of `s` was optional (`String?`), then I'd have to do a nil check before using it.